### PR TITLE
Add "pki/tidy" which allows removing expired certificates.

### PR DESCRIPTION
**Pull request by jefferai**
A buffer is used to ensure that we only remove certificates that are
both expired and for which the buffer has past. Options allow removal
from revoked/ and/or certs/.

### CODE REVIEW

File: builtin/logical/pki/path_tidy.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		},
+
+		HelpSynopsis:    pathTidyHelpSyn,
+		HelpDescription: pathTidyHelpDesc,
+	}
+}
+
+func (b *backend) pathTidyWrite(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	safety_buffer := d.Get("safety_buffer").(int)
```
Comment (vishalnayak) [inline builtin/logical/pki/path_tidy.go]: The documentation says that the default is '72h'. But the data type for safety_buffer is TypeDurationSecond and it takes the value as an integer. User may assume this as a string and supply say '90h' as input. 

Can we mention the data type of fields in both the path-help docs as well as API docs?

File: builtin/logical/pki/path_tidy.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	tidyCertStore := d.Get("tidy_cert_store").(bool)
+	tidyRevocationList := d.Get("tidy_revocation_list").(bool)
+
+	bufferDuration := time.Duration(safety_buffer) * time.Second
+
+	if tidyCertStore {
+		serials, err := req.Storage.List("certs/")
+		if err != nil {
+			return nil, fmt.Errorf("error fetching list of revoked certs: %s", err)
```
Comment (vishalnayak) [inline builtin/logical/pki/path_tidy.go]: these are not revoked certificates

File: builtin/logical/pki/path_tidy.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		}
+
+		for _, serial := range serials {
+			certEntry, err := req.Storage.Get("certs/" + serial)
+			if err != nil {
+				return nil, fmt.Errorf("error fetching certificate %s: %s", serial, err)
+			}
+
+			if certEntry == nil {
+				return nil, fmt.Errorf("revoked certificate entry for serial %s is nil", serial)
```
Comment (vishalnayak) [inline builtin/logical/pki/path_tidy.go]: this is not a revoked certificate

File: builtin/logical/pki/path_tidy.go (hunk 1 above)
Comment (jefferai) [inline builtin/logical/pki/path_tidy.go]: The API docs do mention the data type; I'll fix up path-help. TypeDurationSecond *can* take an integer, but can also take a string -- it always outputs an integer though.

Not accepting it as the default is a bug, but not a user-facing one.

File: builtin/logical/pki/path_tidy.go (hunk 2 above)
Comment (jefferai) [inline builtin/logical/pki/path_tidy.go]: oops, thanks

File: builtin/logical/pki/path_tidy.go (hunk 3 above)
Comment (jefferai) [inline builtin/logical/pki/path_tidy.go]: oops, thanks

**Comment by vishalnayak** [comment]
LGTM! :+1: :ship:

**Pull request merged**